pkg/eval/builtin_fn_str.go

```diff
@@ -49,14 +49,21 @@ func toString(fm *Frame, args ...any) error {
 // greater than 36.
 var ErrBadBase = errors.New("bad base")
 
-func base(fm *Frame, b int, nums ...int) error {
+// ErrInvalidInput is thrown by the "base" builtin if the input num is invalid
+var ErrInvalidInput = errors.New("invalid input")
+
+func base(fm *Frame, b int, nums ...string) error {
```

Comment:
Did you try this: `base 16 (math:pow 2 16)`? Num objects are not automatically converted to strings when calling a builtin, but num objects are automatically converted to int if they fit.

Reply:
Hint: Replace `string` with `vals.Num`. Then add a type switch to allow only int and bigint and reject rational and float64.

Reply:
Thanks for the tip. Adapted in the new version.